components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,84 @@
+const reddit = require("../../reddit.app.js");
+const get = require("lodash.get");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emmits a new link submitted to a subreddit",
+  version: "0.0.2",
+  props: {
+    reddit,
+    timer: {
+      label: "Polling schedule",
+      description:
+        "Pipedream poll for new links and emit accordingly with this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minute
+      },
+    },
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "Enter subreddit name to listen for new links.",
+      default: "redditdev",
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emmits all existing events for the first time.
+      var after = null;
+      do {
+        const reddit_things = await this.reddit.getNewSubredditLinks(
+          null,
+          this.subreddit
+        );
+        if (get(reddit_things, "data.after") !== undefined) {
+          after = reddit_things.data.after;
+          console.log(
+            `after children length: ${reddit_things.data.children.length}`
+          );
+          if (reddit_things.data.children.length > 0) {
```

Comment:
Will `children` always exist, and will it be an empty array, even if Reddit returns no items?

Reply:
Well, I had the issue during troubleshooting and decided to use 
`const get = require("lodash.get");`
just as you suggested a user on the Slack space.
the line
`if (get(reddit_things, "data.after") !== undefined) {`
which is little above the one quoted in this comment should take care of only hitting the line accessing `children`, right?
btw, thank you on the Slack thread about lodash.get)))

Reply:
ok, interestingly there was a case with an issue accessing children, instead I had to check for null values besides undefined. i am doing
`      const links_length = get(reddit_things, "data.children.length");
      var links_pulled =links_length !== null && links_length !== undefined && reddit_things.data.children.length > 0;`

not sure if there is a recommended better practice to handle these cases

Reply:
You have addressed this in a new comment.